### PR TITLE
[WIP] Read ledgers from history archives.

### PR DESCRIPTION
**Pull request by debnil**
WIP PR for reading ledgers from the history archives. I'll fill out the checklist below once the scope of this PR is more finalized.

<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### Summary

[TODO]

### Goal and scope

[TODO]

### Summary of changes

[TODO]

### Known limitations & issues

[TODO]

### What shouldn't be reviewed

[TODO]

### CODE REVIEW

File: exp/ingest/io/archive_ledger_reader.go [hunk 1]
```diff
@@ -1,9 +1,126 @@
...
+
+// Close moves the read pointer so subsequent calls to Read() return EOF.
+func (talr *TransactionsArchiveLedgerReader) Close() error {
+	talr.readMutex.Lock()
+	talr.readIdx = len(talr.transactions)
+	talr.readMutex.Unlock()
+	return nil
+}
+
+// Init pulls data from thee archive backend to set up this object.
```
Comment (abuiles) [inline exp/ingest/io/archive_ledger_reader.go]:
```suggestion
// Init pulls data from the archive backend to set up this object.
```

**Comment by ire-and-curses** [comment]
@cla-bot check

**Comment by leighmcculloch** [comment]
@clabot check

**Comment by debnil** [comment]
Closing - will be superseded by future work from the Horizon team

**Pull request closed without merging**